src/task/task_graph.rs

```diff
@@ -400,35 +416,86 @@ impl<'p> TaskGraph<'p> {
     fn merge_args(
         task_name: &TaskName,
         task_arguments: Option<&Vec<TaskArg>>,
-        arg_values: Option<&Vec<String>>,
+        dep_args: Option<&Vec<TypedDependencyArg>>,
     ) -> Result<ArgValues, TaskGraphError> {
         let task_arguments = match task_arguments {
             Some(args) => args,
             None => &Vec::new(),
         };
 
-        let arg_values = match arg_values {
-            Some(values) => values,
+        let task_arg_names: Vec<String> = task_arguments
+            .iter()
+            .map(|arg| arg.name.as_str().to_owned())
+            .collect();
+
+        let dep_args = match dep_args {
+            Some(args) => args,
             None => &Vec::new(),
         };
 
+        let mut named_args = Vec::new();
+        let mut seen_named = false;
+
+        // build up vec of named args whilst validating that all named args are valid for this task,
+        // and that all positional args precede any named args
+        for arg in dep_args {
+            match arg {
+                TypedDependencyArg::Named(name, value) => {
+                    if !task_arg_names.contains(name) {
+                        // TODO: make an error variant for this
+                        return Err(TaskGraphError::TooManyArguments(format!(
+                            "named argument '{}' does not exist for task {}",
+                            name, task_name
+                        )));
+                    }
+                    seen_named = true;
+                    named_args.push((name.to_string(), value.to_string()));
+                }
+                _ => {
```

Comment:
Use `TypedDependencyArg::Positional` instead of `_` here. That way we get a compiler error if we ever add another variant